docs/onboarding/glossary.md

```diff
@@ -1,7 +1,168 @@
 ---
-id: iron-fish-glossary
-title: Glossary / FAQ
-sidebar_label: Glossary / FAQ
-description: Glossary / FAQ | Iron Fish Onboarding
-hide_table_of_contents: true
+id: glossary
+title: Glossary
+sidebar_label: Glossary
+description: Glossary | Iron Fish Onboarding
+hide_table_of_contents: false
 ---
+
+## Asset
+
+A user-created resource which can be transferred privately, minted, and burned. Contains metadata, a name, the owner public address, and nonce. 
+
+### Asset Generator
+
+The [`group_hash`](#group-hash) of the [Hashed Asset Info](#hashed-asset-info). Calculated in public so the hash can be verified at consensus.
+
+### Asset Identifier
+
+The 32-byte representation of the [Asset Generator](#asset-generator) point.
+
+### Hashed Asset Info
+
+A Pedersen Hash of the Asset Info (owner, name, metadata, nonce) . This hash is the pre-image for the [`group_hash`](#group-hash) to generate the [Asset Generator](#asset-generator) point.
+
+## Block
+
+A piece of data in the Iron Fish chain which holds a record of transactions, graffiti, difficulty, previous [Block](#block), sequence, and timestamp.
+
+### Graffiti
+
+32 byte field that may be assigned at will by the miner who mined the block.
+
+### Difficulty
+
+Measure of how difficult it is to produce an Iron Fish [Block](#block).
+
+### Sequence
+
+The sequential index number of a block in the Iron Fish chain, or the number of blocks preceding a given [Block](#block).
+
+## Circuit
+
+A program used to create the [Proof](#proof) for a [Transaction Description](#description).
+
+## Commitment
+
+Product of a value and random number as a way of committing to a value without revealing the value itself. Generally a point on the elliptic curve.
+
+## Groth16
+
+The zero-knowledge proving system used in Iron Fish and Sapling. Named after the [paper](https://eprint.iacr.org/2016/260) by Jens Groth.
+
+## Group Hash
+
+A function which is used to obtain generator points on the Jubjub curve.
+
+## Keys
+
+A piece of data which can be used to encrypt and decrypt data.
+
+### Spending Key
+
+> Also known as Secret Key
+
+32-byte random seed used to construct other parts of an Iron Fish wallet. The expanded form of this key is required before a note can be spent.
+
+### Spend Authorization Key
+
+> Also known as `ask`
+
+Derived from spending key using a seeded pseudorandom hash function. Used to construct [Authorization Key](#authorization-key).
+
+### Authorization Key
+
+> Also known as `ak`
+
+Public key for the [Spend Authorization Key](#spend-authorization-key) by multiplying the [Spend Authorization Key](#spend-authorization-key) with a fixed generator base point for it on the Jubjub curve.
```

Comment:
```suggestion
Public key constructed by multiplying the [Spend Authorization Key](#spend-authorization-key) with a fixed generator base point on the Jubjub curve.
```

The repeated "Spend Authorization Key" felt like it might be a typo, maybe it could be worded like this?